wordpress-mobile/WordPress-Android
# [Dependency Updates] Reconfigure Media-for-Mobile Dependency

This PR is mainly about reconfiguring the [media-for-mobile](https://github.com/INDExOS/media-for-mobile) dependency, in the sense of extracting its version (5cfcfd1828fc6b8f2d21b8bb292a7dd15bb9fa9b), adding its transitive dependency directly (233ccdee4882375ce63f724f54a9a6815e7572d2) and removing the unused `M4mVideoOptimizer` class (d40542d3dd294230e46017fd517dd854f125c404).

FYI: This [media-for-mobile](https://github.com/INDExOS/media-for-mobile) dependency was added as part of [this PR](https://github.com/wordpress-mobile/WordPress-Android/pull/6088) (see 4a7c9c9e6ffb6645b0795c88553f0f1e23dd476a commit) for video optimization purposes (Cc @daniloercoli). Then, as part of [this PR](https://github.com/wordpress-mobile/WordPress-Android/pull/14256) (see 61ebbbc880fdd73db0b67f7e9ddae4b7794e3904 commit), the unused `M4mVideoOptimizer` implementation was added as well, that is, along side the `Mp4ComposerVideoOptimizer` implementation.

This change will help with maintaining and updating this [media-for-mobile](https://github.com/INDExOS/media-for-mobile) dependency. Also, opening this PR gives me the opportunity to ask a couple of questions about this library, its maintenance and implementation overall:

1. @daniloercoli since you were the last to contribute to the [media-for-mobile](https://github.com/INDExOS/media-for-mobile/commits/master) project (`Jun 15, 2017`), what's the state of this project, should it be considered outdated, should we start maintaining it more actively. Also, this project is under the [INDExOS](https://github.com/INDExOS) umbrella, I am not sure what that means, can you shed some light on that too? 🤔
2. @develric since you were the one that added this unused [M4mVideoOptimizer](https://github.com/wordpress-mobile/WordPress-Android/blob/a2218f0ea74ff4ee996b6266f09b290c43bc6aa1/WordPress/src/main/java/org/wordpress/android/ui/uploads/M4mVideoOptimizer.java) implementation. Is that really as unused as it seems, is there a reason for that, is it safe to remove this implementation? 🤔

----

PS: @develric I added you as the main reviewer, that is, in addition to the @wordpress-mobile/apps-infrastructure  team itself, but not entirely randomly as I am seeing you having some experience on working with video optimization, and want someone from the WordPress team to be aware of and sign-off on that change for WPAndroid.

----

To test:

1. See the dependency tree diff result and verify correctness.
2. Smoke test any video optimization related functionality on both, the WordPress and Jetpack apps, and see if they both work as expected. For more info on how to test the video optimization you can follow the testing instructions on this #14256 PR.

----

## Regression Notes

1. Potential unintended areas of impact

    - The `media-for-mobile:domain` transitive dependency added might be causing some kind of misbehaviour.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

PR submission checklist:
